src/calculateRentalCost.js

```diff
@@ -7,4 +7,24 @@
   // write code here
 }
 
+const DAILY_RATE = 40;
+const LONG_TERM_RENTAL_DAYS = 7;
+const LONG_TERM_DISCOUNT = 50;
+const MID_TERM_RENTAL_DAYS = 3;
+const MID_TERM_DISCOUNT = 20;
+
+function calculateRentalCost(numberOfDays) {
+  const baseCost = numberOfDays * DAILY_RATE;
+
+  if (numberOfDays >= LONG_TERM_RENTAL_DAYS) {
+    return baseCost - LONG_TERM_DISCOUNT;
+  }
+
+  if (numberOfDays >= MID_TERM_RENTAL_DAYS) {
+    return baseCost - MID_TERM_DISCOUNT;
+  }
```

Comment:
The current logic applies only one discount based on the rental period. If a rental period qualifies for the long-term discount, it should not check for the mid-term discount. Consider restructuring the logic to ensure that the correct discount is applied based on the number of days.